### ISSUE TITLE
Add shortcut to Navigation by Annotations

### ISSUE THREAD

**Issue by gautierchomel** [edrlab/thorium-reader#2730]
Some navigation menus can be reached by a keyboard shortcut, like "FocusReaderGotoPage" and "FocusReaderNavigationTOC". It is not the case for Navigation by Annotations. 

The Annotation panel contains a button per annotation that will bring the Focus to the Annotation in the content, leaving the Annotation panel context. To get back to the list of annotations, keyboard users have to tabulate various times. 

Adding a keyboard shortcut to reach the Annotation navigation panel quickly would remove the complexity of the user experience while consulting annotations.

**Comment by gautierchomel** [comment]
This also apply to bookmarks

**Comment by gautierchomel** [comment]
This appears to be implemented but non funcional (macOS and Debian).

https://github.com/edrlab/thorium-reader/blob/c2140af251aa524cd8bb8f1f2a023ea34017ce45/src/common/keyboard.ts#L123-L129

But Bookmarks works 

https://github.com/edrlab/thorium-reader/blob/c2140af251aa524cd8bb8f1f2a023ea34017ce45/src/common/keyboard.ts#L115-L121

**Comment by gautierchomel** [comment]
Changing the shortcut does not work either.

**Comment by gautierchomel** [comment]
The shortcut is not localised. Works in qwerty, not in Azerty.

**Comment by gautierchomel** [comment]
The problem appears in 3.3 but not in develop.

**Comment by danielweck** [comment]
Ah, actually the problem is more subtle: I tested on MacOS with the official DMG `3.3.0` and the latest automated build `3.3.1-beta` => the bug is 100% reproducible. The bug is: the keyboard shortcut editor displays the key characters always in the context of QWERTY instead of switching to the correct locale, for example AZERTY.

HOWEVER, if I run the exact same code (`3.3.1-beta`) locally from a developer build (no DMG packaging), the feature works absolutely fine: I can select "British / QWERTY" (default on my Mac laptop) or "French / AZERTY" in the system-level preferences, and Thorium's keyboard shortcut editor correctly shows the inverted `KeyA` / `KeyB` (which are located immediately next on top of each other, physically on the keyboard).

**Comment by danielweck** [comment]
PS: no need to restart Thorium or to re-open the application preferences dialog, simply switch to a different tab header and come back to keyboard shortcuts.

Relevant code:

https://github.com/edrlab/thorium-reader/blob/a068dc807b216ebb42c2ce4f275d0b1f70404209/src/renderer/library/components/settings/KeyboardSettings.tsx#L188-L190

https://github.com/edrlab/thorium-reader/blob/a068dc807b216ebb42c2ce4f275d0b1f70404209/src/renderer/library/components/settings/KeyboardSettings.tsx#L202-L206

https://github.com/edrlab/thorium-reader/blob/a068dc807b216ebb42c2ce4f275d0b1f70404209/src/renderer/library/components/settings/KeyboardSettings.tsx#L959-L964

https://github.com/edrlab/thorium-reader/blob/a068dc807b216ebb42c2ce4f275d0b1f70404209/src/renderer/library/components/settings/KeyboardSettings.tsx#L210-L235

https://github.com/edrlab/thorium-reader/blob/a068dc807b216ebb42c2ce4f275d0b1f70404209/src/renderer/library/components/settings/KeyboardSettings.tsx#L843-L853

https://github.com/edrlab/thorium-reader/blob/a068dc807b216ebb42c2ce4f275d0b1f70404209/src/renderer/library/components/settings/KeyboardSettings.tsx#L992-L1002

**Comment by danielweck** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/Keyboard/getLayoutMap

> Experimental: This is an experimental technology. Check the Browser compatibility table carefully before using this in production.
> Secure context: This feature is available only in secure contexts (HTTPS), in some or all supporting browsers.

**Comment by danielweck** [comment]
> SecurityError DOMException
> Thrown if the call is blocked by a "permission policy"

https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/Permissions_Policy

> Experimental: This is an experimental technology
> Check the Browser compatibility table carefully before using this in production.

**Comment by danielweck** [comment]
So, `navigator.keyboard.getLayoutMap()` was the solution to the problem of QWERTY-bound keyboard shortcuts defined by Thorium Desktop not being shown to the user with the correct character to match their keyboard layout. A typical example is `SHIFT OPT/ALT Q` for "Quick annotation" is in fact `A` on a French AZERTY keyboard, and vice-versa `SHIFT OPT/ALT A` for "Create annotation with editor" is in fact `Q`!! ... very confusing for a French user as the annotation feature seems to work, but not quite correctly! Besides the inverted `Q` and `A` there are other much more problematic cases where the "localised" keys are not co-located on the keyboard, thus why it was very important to find a solution ... `navigator.keyboard.getLayoutMap()` seemed to fit the bill perfectly, but unfortunately it appears to be unreliable (and it is clearly marked as experimental anyway).

Ultimately I think Thorium Desktop keyboard shortcuts should be expressed with characters independent from the key layout, not bound to QWERTY. For example on MacOS `CMD Q` is always `Q` to "Quit", regarless of the keyboard layout and locale. Unless I am mistaken, this is the case on French AZERTY, right @gautierchomel ? (which is convenient because "Quitter" starts with the same letter, easy to remember ... not sure about other language though, is it always `Q`?)

**Comment by danielweck** [comment]
https://www.toptal.com/developers/keycode
